tests/providers/snowflake/hooks/test_snowflake.py

```diff
@@ -134,3 +136,15 @@ def test_key_pair_auth_not_encrypted(self):
         self.conn.password = None
         params = self.db_hook._get_conn_params()
         self.assertTrue('private_key' in params)
+
+    def test_authenticator(self):
+        self.conn.extra_dejson = {'database': 'db',
+                                  'account': 'airflow',
+                                  'warehouse': 'af_wh',
+                                  'region': 'af_region',
+                                  'role': 'af_role',
+                                  'authenticator': 'externalbrowser'}
+
+        uri_shouldbe = 'snowflake://user:pw@airflow/db/public?warehouse=af_wh&role=af_role' \
+                       '&authenticator=externalbrowser'
+        self.assertEqual(uri_shouldbe, self.db_hook.get_uri())
```

Comment:
This test is not related to the changes in this PR. It just checks if `extra_dejson` of any Connection object is changed, they are reflected and can be retrieved with `get_uri()`.